src/object/cli_obj.c

```diff
@@ -4247,12 +4274,13 @@ dc_obj_fetch_task(tse_task_t *task)
 		obj_auxi->to_leader = (args->extra_flags & DIOF_TO_LEADER) != 0;
 	}
 
-	if (obj_auxi->csum_retry && !obj_auxi->is_ec_obj) {
-		rc = obj_retry_csum_err(obj, obj_auxi, dkey_hash, map_ver,
-					&csum_bitmap);
+	if ((obj_auxi->csum_retry || obj_auxi->tx_uncertain) &&
+	    !obj_auxi->is_ec_obj) {
+		rc = obj_retry_next_shard(obj, obj_auxi, dkey_hash, map_ver);
```

Comment:
just confirm that for EC obj, it still possibly to hit the uncertain case right? 
will it just retry the EC obj IO? is it possible retry forever or for long time?

Reply:
For EC object, it is possible to hit uncertain case, but we cannot simply retry as for replicated object, because the next shard does not contain the expected data, instead, we need to degraded fetch under such case, similar process as the checksum error on a data shard for an EC object.

The uncertain case only happened when the leader and non-leader out-of-sync. It is relative rare, especially multiple replicas are all out-of-sync with the leader. So it will retry until hit one healthy non-leader or leader, will not forever, and very rare for very long time.

Reply:
OK, agreed that "we need to degraded fetch under such case, similar process as the checksum error on a data shard for an EC object". That can be done at a following patch.